hack/dashboard/openshift/deploy-grafana.sh

```diff
@@ -1,56 +1,240 @@
+		info "cluster:  $(oc whoami --show-server)"
+		ret=1
+	}
+
+	oc get crds | grep kepler || {
+		fail "Missing Kepler CRD. Is Kepler Operator Installed?"
+		ret=1
+	}
+
+	oc get kepler kepler || {
+		fail "Mising kepler deployment. Did you create kepler instance?"
+		ret=1
+	}
+
+	return $ret
+}
+
+enable_userworkload_monitoring() {
+	header "Enabling Openshift User Project Monitoring"
+
+	info "checking if user project monitoring is already enabled"
+	[[ $(oc get prometheus -n $UWM_NS -o name | wc -l) -gt 0 ]] && {
+		ok "User project monitoring already enabled; no further action is required"
+		return 0
+	}
+
+	backup_monitoring_config
+
+	if [[ -f "$BACKUP_CMO_CFG" ]]; then
+		patch_enable_uwm
+	else
+		setup_new_uwm
+	fi
+	show_restore_info
+
+	wait_until 10 10 "$UWM_NS to be created " oc get ns "$UWM_NS"
+	wait_until 10 10 "User Workload Prometheus to be created" \
+		oc get -n "$UWM_NS" prometheus user-workload
+}
+
+patch_enable_uwm() {
+	info "Patching existing config to enable User Project Monitoring"
+	line 60
+
+	if grep enableUserWorkload "$BACKUP_CMO_CFG"; then
+		info "toggling enableUserWorkload to true"
+		sed -e 's|enableUserWorkload:.*|enableUserWorkload: true 👈 |g' <"$BACKUP_CMO_CFG"
+
+		sed -e 's|enableUserWorkload:.*|enableUserWorkload: true|g' <"$BACKUP_CMO_CFG" |
+			oc apply --server-side --force-conflicts -f-
+
+	else
+		info "adding enableUserWorkload to true"
+		sed \
+			-e 's|\(config.yaml.*\)|\1\n    enableUserWorkload: true 👈\r|g' \
+			"$BACKUP_CMO_CFG"
+
+		sed \
+			-e 's|\(config.yaml.*\)|\1\n    enableUserWorkload: true\r|g' \
+			"$BACKUP_CMO_CFG" |
+			oc apply --server-side --force-conflicts -f-
+	fi
+
+}
+
+setup_new_uwm() {
+	info "Creating new config to enable User Project Monitoring"
+	line 60
+	cat $CMO_ENABLE_UWM_CFG
+	line 60
+	run oc apply -n "$MON_NS" -f "$CMO_ENABLE_UWM_CFG"
+}
+
+show_restore_info() {
+	line 60 heavy
+	warn "😱 In the event cluster-monitoring-operator becomes 👉 degraded  😱"
+	cat <<-EOF
+
+		  * Restore the configuration $BACKUP_CMO_CFG 
+		  * Manually enable User Project Monitoring 
+			  💡 see: https://docs.openshift.com/container-platform/latest/monitoring/enabling-monitoring-for-user-defined-projects.html
+		  * Rerun this script
+
+	EOF
+	line 60 heavy
+	sleep 10
+}
+
+backup_monitoring_config() {
+	info "Backing up current monitoring configuration"
+
+	oc get -n "$MON_NS" configmap "$CMO_CM" || {
+		info "No monitoring config exists; will create a new one"
+		return
+	}
+
+	# backup but remove last-applied-configuration
+	oc get -n "$MON_NS" configmap "$CMO_CM" -o yaml |
+		grep -Ev 'kubectl.kubernetes.io/last-applied-configuration|{"apiVersion":|creationTimestamp:|resourceVersion:|uid:' >"$BACKUP_CMO_CFG" || {
+		fail "failed to backup $CMO_CM in $MON_NS"
+		return 1
+	}
+
+	ok "Backed up $CMO_CM config to $BACKUP_DIR"
+}
+
+setup_grafana() {
+	header "Setting up Grafana"
+
+	info "Creating Grafana resources"
+	oc_apply_kepler_ns -k hack/dashboard/openshift/grafana-deploy
```

Comment:
Is there any specific reason to deploy grafana inside openshift-kepler-operator ns? Shouldn't we isolate this?

Reply:
I don't have a strong opinion on this. The intention is that when `kepler` is removed, the ns is removed and thus grafana is removed as well.